.github/actions/has-auto-releasable-commits-v1/src/run.test.ts

```diff
@@ -0,0 +1,288 @@
+import 'mocha'
+import { assert } from 'chai'
+import sinon from 'sinon'
+import type { Core } from './types'
+import run from './run'
+
+describe('run', () => {
+  let setFailed: sinon.SinonSpy
+  let setOutput: sinon.SinonSpy
+  let getInput: sinon.SinonStub
+
+  beforeEach(() => {
+    setFailed = sinon.spy()
+    setOutput = sinon.spy()
+    getInput = sinon.stub()
+  })
+
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  describe('when commits input is not provided', () => {
+    it('should throw an error', async () => {
+      getInput.withArgs('commits', { required: true }).throws({
+        message: 'Input required and not supplied: commits'
+      })
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(
+        setFailed.calledWith('Input required and not supplied: commits')
+      )
+    })
+  })
+
+  describe('when version-locked input is not provided', () => {
+    it('should throw an error', async () => {
+      getInput.withArgs('commits', { required: true }).returns('[]')
+      getInput.withArgs('version-locked', { required: true }).throws({
+        message: 'Input required and not supplied: version-locked'
+      })
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(
+        setFailed.calledWith('Input required and not supplied: version-locked')
+      )
+    })
+  })
+
+  describe('when version-locked input is not true or false', () => {
+    it('should throw an error', async () => {
+      getInput.withArgs('commits', { required: true }).returns('[]')
+      getInput.withArgs('version-locked', { required: true }).returns('foo')
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(
+        setFailed.calledWith(
+          'Invalid value for version-locked: foo. Must be true or false'
+        )
+      )
+    })
+  })
+
+  describe('when version-locked input is false', () => {
```

Comment:
See comment about using a testing matrix (as well as taking into account `fix`, `feat`, `fix!` and `feat!`)